migrate/20241204_vm_enforce_vcpus_memory.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    # Update the memory & vcpus for exisitng VMs, to match the logic from the
+    # code. Then enforce the NOT NULL constraint on the columns.
+    run <<-SQL
+    UPDATE vm
+    SET
+      vcpus = CASE
+        WHEN arch = 'arm64' THEN cores
+        ELSE 2 * cores
+      END,
+      memory_gib = CASE
+        WHEN arch = 'arm64' THEN 3.2 * cores
+        WHEN family = 'standard-gpu' THEN 10.68 * cores
+        ELSE 8 * cores
+      END;
+
+    ALTER TABLE vm
+      ALTER COLUMN vcpus SET NOT NULL,
+      ALTER COLUMN memory_gib SET NOT NULL;
+
+    SQL
```

Comment:
This one should change the cache a second time, right? That's where the NOT NULL comes in.

Also in this case you probably want an `up` block, `change` blocks are really for sequel auto-generating the up/down migration.

Reply:
I think that means that those two migrations should be put as separate PRs.

Reply:
@pykello - also, since you split it into separate transactions/commits, I think the UPDATE statement should only touch the columns that are NULL.

Reply:
> Also in this case you probably want an up block, change blocks are really for sequel auto-generating the up/down migration.

@fdr Changed to `up`. Also included a `down` block.

> I think that means that those two migrations should be put as separate PRs.

@macieksarnowicz I'm not sure why? Migrations that can be deployed together are usually grouped in the same commit.

Applied the remaining points.

Reply:
I am assuming the cache has to be updated for each migration separately, that's why.